### ISSUE TITLE
Ticking Memory Crash

### ISSUE THREAD

**Issue by wizard1399** [Mithion/ArsMagica2#1543]
The current version of the mod causes a ticking memory crash any time a potion effect is applied to the player. I am creating a mod pack and have specifically gone through and changed all the potion ids manually, so I know it is not a id conflict.  Could someone look through the code and check the potion listeners and such?

**Comment by ThatOneTraitor** [comment]
Crash logs needed, forge and client

**Comment by wizard1399** [comment]
[Crash Report](http://pastebin.com/YnhRS22w)
[Latest FML log](http://pastebin.com/XWGD78z9)
[Potion Dump](http://pastebin.com/90ZvJd6E)

When I revert to version 1.4.0.008 I do not get this crash and everything works fine. I get this crash when any potion effect is applied, even vanilla minecraft potions. I am using extended potions, but I can confirm that it is not extended potions causing the crash. If it were extended potions I would get the crash on ver .008 as well. I have included the potion data dump as well. The AM2 potions are at the bottom of the list. The config was changed manually to set the potion ids to those found in the dump. I can confirm that the id's in the dump and config match.

**Comment by renadi** [comment]
The crash here is caused by a thrown potion in the world, remove erroring
entities in forge setting and it should work I think.

This usually occurs if you change potion ids mid game, so there is a potion
that thinks it should be two different potions.

I believe if you just select that option from the forge configs and reload
it'll delete that potion and any new ones will be assigned the new id.

**Comment by wizard1399** [comment]
I am not sure this will fix it. I used a splash potion this time to check if vanilla potions caused the crash as well. Let me try a test. I'll start a brand new world and try some different potion effects to see if I still get the crash.

**Comment by wizard1399** [comment]
Any news on this issue?

**Comment by Iivaitte** [comment]
Have you tried without ACE?

**Comment by wizard1399** [comment]
I have not. Did not even consider that as no potion effects are applied through ace. Will test and get back to you.

**Comment by wizard1399** [comment]
I tried taking ACE out of the pack and then applying a potion effect. I still get a crash. What I can't figure out is why i get the crash with ver .009 and not with .008. Since I can't find a change log to figure out what got changed and am not knowledgeable enough of how this mod's code is working, I can't offer much help. Could one of the mod authors review the changes that were made to try and figure out what is causing the crash?

**Comment by ThatOneTraitor** [comment]
```
-- Entity being ticked --
Details:
    Entity Type: ThrownPotion (net.minecraft.entity.projectile.EntityPotion)
    Entity ID: 1272
    Entity Name: entity.ThrownPotion.name
    Entity's Exact location: -268.29, 86.14, 884.26
    Entity's Block location: World: (-269,86,884), Chunk: (at 3,5,4 in -17,55; contains blocks -272,0,880 to -257,255,895), Region: (-1,1; contains chunks -32,32 to -1,63, blocks -512,0,512 to -1,255,1023)
    Entity's Momentum: -0.00, -0.46, -0.00
Stacktrace:
    at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:489)
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:636)
```

**Comment by wizard1399** [comment]
The potion in question was pulled directly from the creative tab in a test world. Any potion affect applied by any means causes a crash wether it be a vanilla potion affect applied by thrown or drunken or a modded potion affect.

**Comment by bookerthegeek** [comment]
Do me a favor? Remap the potions to be 127 or below. Do you still crash?

**Comment by wizard1399** [comment]
Let me do some reconfiging ang report back.

**Comment by wizard1399** [comment]
Fresh potion dump using NEI tools:
[potion.txt](https://github.com/Mithion/ArsMagica2/files/402966/potion.txt)

AM2 potions start at ID 24

After testing, I still get the crash.

**Comment by wizard1399** [comment]
So after doing a good bit of testing and research, I have found that this is a mod conflict caused by DragonAPI.  Here are some of the things I was able to find out:
Forge loads core mods first in alphabetical order, then the rest of the mods.
AM2 preloader is regularly the first core mod to load.
AM2 [pull request 1322](https://github.com/Mithion/ArsMagica2/pull/1322) changed how AM2 extends the potion array by use of byte transforms.
As mentioned in the PR1322 the potion array is not extended if DragonAPI is detected.
From what I can tell Forge does not recognize DragonAPI as a core mod.
When I took DragonAPI out, and all associated mods that require it as a prerequisite, potions effects seem to be applied correctly. 
When I looked through the issues for AM2, whenever the issue involved is potion affects, DragonAPI is listed in the crash report as one of the mods installed. 

So some questions for both mod authors, I will reference this issue over in Reikka's forum.
Is this a matter of what forge loads first? 
Is there a fix or workaround that can be coded in?
Are AM2 and DragonAPI simply incompatible?
